### ISSUE TITLE
spm build: TypeError: add() got an unexpected keyword argument 'filter'

### ISSUE THREAD

**Issue by deutsche** [saltstack/salt#29263]
spm --version
spm 2015.8.1 (Beryllium)

python 2.6 does not support filter argument. Tested on Centos 6.
```
[root@centos6]# spm build apache-formula/
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
TypeError: add() got an unexpected keyword argument 'filter'
Traceback (most recent call last):
  File "/usr/bin/spm", line 12, in <module>
    salt_spm()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 475, in salt_spm
    spm.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/spm.py", line 32, in run
    client.run(self.args)
  File "/usr/lib/python2.6/site-packages/salt/spm/__init__.py", line 103, in run
    self._build(args)
  File "/usr/lib/python2.6/site-packages/salt/spm/__init__.py", line 572, in _build
    formula_tar.add(self.abspath, formula_conf['name'], filter=self._exclude)
TypeError: add() got an unexpected keyword argument 'filter'
Traceback (most recent call last):
  File "/usr/bin/spm", line 12, in <module>
    salt_spm()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 475, in salt_spm
    spm.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/spm.py", line 32, in run
    client.run(self.args)
  File "/usr/lib/python2.6/site-packages/salt/spm/__init__.py", line 103, in run
    self._build(args)
  File "/usr/lib/python2.6/site-packages/salt/spm/__init__.py", line 572, in _build
    formula_tar.add(self.abspath, formula_conf['name'], filter=self._exclude)
TypeError: add() got an unexpected keyword argument 'filter'
```
from https://docs.python.org/2/library/tarfile.html#tarfile.TarFile.add
```
Deprecated since version 2.7: The exclude parameter is deprecated, please use the filter parameter instead. For maximum portability, filter should be used as a keyword argument rather than as a positional argument so that code won’t be affected when exclude is ultimately removed.
```

**Comment by techhat** [comment]
Thanks for the report @deutsche. This was fixed in #28521.

**Comment by cachedout** [comment]
It defeinitely looks like this fix should address the problem. I'm going to go ahead and close this.
